### PR TITLE
[3006.x] Fix issue with win_user.add

### PR DESCRIPTION
**Pull request by twangboy**
### What does this PR do?
`win_useradd.add` now allows you to add new users with only integers in the username.
It also does the following:
- Adds tests for the win_useradd module
- Removes the single reference to `__utils__` in the module
- Fixes a bug setting the account expiration date (found in testing)
- Fixes a bug using `append=True` when adding groups for the user
- 

### What issues does this PR fix or reference?
Fixes - #53363 

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [x] Docs
- [x] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [x] Tests written/updated

### Commits signed with GPG?
Yes

### CODE REVIEW

File: tests/pytests/unit/modules/test_win_useradd.py [hunk 1]
```diff
@@ -0,0 +1,344 @@
...
+@pytest.fixture
+def username_str():
+    _username = random_string("test-account-", uppercase=False)
+    try:
+        yield _username
+    finally:
+        try:
+            user.delete(_username, purge=True, force=True)
+        except Exception:  # pylint: disable=broad-except
+            # The point here is just system cleanup. It can fail if no account was created
```
Comment (s0undt3ch) [inline tests/pytests/unit/modules/test_win_useradd.py:30]: If we actually create accounts, then move this to functional please.

File: tests/pytests/unit/modules/test_win_useradd.py [hunk 2]
```diff
@@ -0,0 +1,344 @@
...
+    assert salt.utils.data.compare_lists(ret["groups"], groups) == {}
+
+
+def test_chgroups_single_int(account_int):
+    groups = ["Backup Operators"]
+    ret = user.chgroups(account_int.username, groups=groups)
+    assert ret is True
+    ret = user.info(account_int.username)
+    groups.append("Users")
+    assert salt.utils.data.compare_lists(ret["groups"], groups) == {}
```
Comment (s0undt3ch) [inline tests/pytests/unit/modules/test_win_useradd.py:116]: Why are we using this compare list salt utility?

Couldn't we compare against the sorted lists?

File: tests/pytests/unit/modules/test_win_useradd.py [hunk 3]
```diff
@@ -0,0 +1,344 @@
...
+
+
+def test_delete_int(account_int):
+    ret = user.delete(name=account_int.username)
+    assert ret is True
+    assert user.info(name=account_int.username) == {}
+
+
+def test_getUserSig_str(account_str):
+    ret = user.getUserSid(account_str.username)
```
Comment (s0undt3ch) [inline tests/pytests/unit/modules/test_win_useradd.py:198]: Why did we add a camel cased function name to a salt module I  the first place?!?!

**Review by s0undt3ch — DISMISSED** [review]

File: tests/pytests/unit/modules/test_win_useradd.py (hunk 1 above)
Comment (twangboy) [reply tests/pytests/unit/modules/test_win_useradd.py:30]: Functional was a lot slower

File: tests/pytests/unit/modules/test_win_useradd.py (hunk 2 above)
Comment (twangboy) [reply tests/pytests/unit/modules/test_win_useradd.py:116]: The comparison didn't work when I was using assert list1 == list2... I'll try it again

Comment (twangboy) [reply tests/pytests/unit/modules/test_win_useradd.py]:
```
AssertionError: assert ['Users', 'Backup Operators', 'Guests'] == ['Backup Operators', 'Guests', 'Users']
```

File: tests/pytests/unit/modules/test_win_useradd.py (hunk 1 above)
Comment (twangboy) [reply tests/pytests/unit/modules/test_win_useradd.py]: Moved to functional, but kept the `configure_loader_modules` fixture

File: tests/pytests/unit/modules/test_win_useradd.py (hunk 2 above)
Comment (twangboy) [reply tests/pytests/unit/modules/test_win_useradd.py]: Used `groups.sort()` and `ret["gruops"].sort()` and removed the `salt.utils.data` import

File: tests/pytests/functional/modules/test_win_useradd.py [hunk 4]
```diff
@@ -0,0 +1,343 @@
...
+    pytest.mark.skip_unless_on_windows,
+    pytest.mark.windows_whitelisted,
+]
+
+
+@pytest.fixture
+def configure_loader_modules():
+    return {user: {"__salt__": {"cmd.run_all": salt.modules.cmdmod.run_all}}}
+
+
```
Comment (s0undt3ch) [inline tests/pytests/functional/modules/test_win_useradd.py:19]:
```suggestion
```
You don't need this anymore

**Review by s0undt3ch — DISMISSED** [review]

File: tests/pytests/functional/modules/test_win_useradd.py (hunk 4 above)
Comment (twangboy) [reply tests/pytests/functional/modules/test_win_useradd.py]: OK, removed it and the direct module imports. Created a user fixture that returns `modules.user`.

File: tests/pytests/functional/modules/test_win_useradd.py [hunk 5]
```diff
@@ -0,0 +1,341 @@
...
+    assert ret["fullname"] == "New Full Name"
+
+
+def test_chgroups_single_str(user, account_str):
+    groups = ["Backup Operators"]
+    ret = user.chgroups(account_str.username, groups=groups)
+    assert ret is True
+    ret = user.info(account_str.username)
+    groups.append("Users")
+    assert ret["groups"].sort() == groups.sort()
```
Comment (s0undt3ch) [inline tests/pytests/functional/modules/test_win_useradd.py:104]:
```suggestion
    assert sorted(ret["groups"]) == sorted(groups)
```

`.sort()` does not return anything, while sorted does.


```python
>>> a = [3, 1, 0]
>>> b = [4, 1, 2]
>>> a.sort()
>>> sorted(b)
[1, 2, 4]
```

If you really wanted to use `.sort()`
```python
    ret["groups"].sort()
    groups.sort()
    assert ret["groups"] == groups
```

**Review by s0undt3ch — DISMISSED** [review]

File: tests/pytests/functional/modules/test_win_useradd.py (hunk 5 above)
Comment (twangboy) [reply tests/pytests/functional/modules/test_win_useradd.py:104]: No wonder they all passed...

Comment (twangboy) [reply tests/pytests/functional/modules/test_win_useradd.py]: Changed to use sorted instead of sort

**Review by s0undt3ch — DISMISSED** [review]

**Review by dmurphy18 — APPROVED** [review]

**Review by s0undt3ch — APPROVED** [review]

**Pull request merged**